mlflow/transformers.py

```diff
@@ -2051,12 +2072,12 @@ def _parse_lists_of_dict_to_list_of_str(self, output_data, target_dict_key) -> L
                         elif isinstance(value, list) and all(
                             isinstance(elem, dict) for elem in value
                         ):
-                            output_coll.append(
-                                self._parse_lists_of_dict_to_list_of_str(value, target_dict_key)[0]
+                            output_coll.extend(
+                                self._parse_lists_of_dict_to_list_of_str(value, target_dict_key)
                             )
                 elif isinstance(output, list):
-                    output_coll.append(
-                        self._parse_lists_of_dict_to_list_of_str(output, target_dict_key)[0]
+                    output_coll.extend(
+                        self._parse_lists_of_dict_to_list_of_str(output, target_dict_key)
```

Comment:
I changed this because for example if you set top_k=2, the previous method only fetch the first result returned. I feel like we should just extend whatever results are returned.